plugin/pkg/scheduler/modeler.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	algorithm "github.com/GoogleCloudPlatform/kubernetes/pkg/scheduler"
+
+	"github.com/golang/glog"
+)
+
+var (
+	_ = SystemModeler(&FakeModeler{})
+	_ = SystemModeler(&SimpleModeler{})
+)
+
+// ExtendedPodLister: SimpleModeler needs to be able to check for a pod's
+// existance in addition to listing the pods.
+type ExtendedPodLister interface {
+	algorithm.PodLister
+	Exists(pod *api.Pod) bool
+}
+
+// FakeModeler implements the SystemModeler interface.
+type FakeModeler struct {
+	AssumePodFunc func(pod *api.Pod)
+}
+
+// AssumePod calls the function variable if it is not nil.
```

Comment:
`go vet` again... Yeah, I don't know, I try to follow 'godoc style' which requires a comment of this form for all public items. Will leave unless you feel realllly strongly.